src/git/github_api_extension.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python
+# Copyright OpenSearch Contributors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Extension to GitRepository for GitHub API integration."""
+
+import logging
+import time
+from typing import Dict, List, Optional
+
+import requests
+
+from git.git_repository import GitRepository
+
+
+class GitRepositoryWithGitHubAPI(GitRepository):
+    """Extends GitRepository with GitHub API capabilities."""
+    
+    def __init__(self, url: str, ref: str, directory: str = None, working_subdirectory: str = None, github_token: str = None):
+        super().__init__(url, ref, directory, working_subdirectory)
+        
+        self.github_token = github_token
+        self.headers = {"Authorization": f"token {github_token}"} if github_token else {}
+        self.session = requests.Session()
+        if github_token:
+            self.session.headers.update(self.headers)
+        
+        # Extract repo name from URL
+        self.repo_name = url.rstrip('/').split('/')[-1].replace('.git', '')
+    
+    def get_changelog_from_github(self) -> Optional[str]:
+        """Get CHANGELOG.md content from GitHub API."""
+        if not self.github_token:
+            logging.warning("No GitHub token provided, cannot fetch changelog via API")
```

Comment:
Would the warning logger be displayed during normal run mode or would it be only for debug mode?